docs/design/proposals/logical-volume-manager-storage-api-v2.md

```diff
@@ -0,0 +1,591 @@
+
+    LVMNodeSetBeta[v2alpha1/LVMNodeSet β]
+    LVMNodeBeta1[v2alpha1/LVMNode β 1]
+
+    NodeN1[v1/Node N1]
+    NodeN2[v1/Node N2]
+    NodeN2[v1/Node N3]
+
+    LVMSAdmin-->|Creates|LVMNodeSetAlpha;
+    LVMSAdmin-->|Creates|LVMNodeSetBeta;
+
+    LVMNodeSetAlpha-->|OwnedBy|LVMNodeAlpha1;
+    LVMNodeSetAlpha-->|OwnedBy|LVMNodeAlpha2;
+    LVMNodeAlpha1-->|sync DeviceClasses and VGs on host|NodeN1;
+    LVMNodeAlpha2-->|sync DeviceClasses and VGs on host|NodeN2;
+    LVMNodeSetBeta-->|OwnedBy|LVMNodeBeta1;
+    LVMNodeBeta1-->|sync DeviceClasses and VGs on host|NodeN3;
+```
+
+1. LVMS Administrator installs LVMS through ClusterServiceVersion
+2. LVMS Administrator creates any combination of `LVMNodeSet` with NodeSelector and Template for VolumeGroups and DeviceClasses based on the cluster topology.
+3. `LVMNodeProviderReconciler` verifies we have a provider (e.g. TopoLVM) for the LVMNode in the cluster and installs it if necessary
+4. If the provider is ready, `LVMNode.Status.Conditions[].(type=ProviderReady)` is set to `True`
+5. As soon as the provider is ready, `LVMNode` is picked up by`vgmanager`.
+6. `LVMNode.spec.volumeGroups` are created and reconciled by `vgmanager`.
+7. `LVMNode.status.volumeGroups` are synchronized and `LVMNode.Status.Conditions[].(type=VolumeGroupsHealthy)` is set to `True`
+8. `LVMNode.spec.deviceClasses` are created (with thin-pool LV on top) and synchronized to `lvmd.conf` by `vgmanager`.
+9. `StorageClass` and `SnapshotClass` get created for DeviceClass by `vgmanager`.
+10. `LVMNode.status.deviceClasses` are synchronized and `LVMNode.Status.Conditions[].(type=DeviceClassesHealthy)` is set to `True`
+11. `LVMNode.status.state` is set to `Ready` after all conditions succeed.
+12. `LVMNodeSetReconciler` picks up the `Ready` condition and sets `LVMNodeSet.Status.Conditions[].(type=LVMNodesReady)` to `True`
+13. `LVMNodeSet.status.state` is set to `Ready` after all conditions succeed.
+
+Contrasting this to the old workflow, we have significantly easier resource tracking ability.
+This is because now, `.spec` and `.status` correlate with each other for both `LVMNode` and `LVMNodeSet`. Previously,
+one had to look at `LVMCluster`, `LVMVolumeGroup` and `LVMVolumeGroupNodeStatus` as well as logs to get detailed information on the `lvm` subsystem.
+With this proposed change, it is easily possible to recognize the `lvm` subsystem for single nodes by looking at `LVMNode` alone.
+
+### API Extensions
+
+#### Summary
+
+1. The usage of a `LVMNodeSet` akin to `DaemonSet`. This is used to align the different `Node`'s into one single configurable object.
+2. The introduction of a `spec.template`, similar to a `Deployment`. This template will be used by all nodes in the `LVMNodeSet` to configure individual nodes.
+3. Generic Conditions for checking if the LVMNodeSetup for all nodes is ready.
+4. Separation of `deviceClasses` from `volumeGroups`. This is because VolumeGroups in theory can be reused by different deviceClasses
```

Comment:
I think the benefits of this separation need more clarification. The only argument here is `VolumeGroups in theory can be reused by different deviceClasses`, but we also say `Important to note is that any single deviceClass can only be mapped to one volumeGroup.` below. Does this mean 
- device classes will be 1:1 matching with volume groups? If so, what would be the other benefits of this separation?
- multiple device classes can use a single volume group?  If so, what would happen when a thin-typed device class and thick-typed one try to use the same volume group, will this be allowed?

Reply:
Basically this means that one deviceClass can be mapped to n volume groups, but there can be at most one volume group assigned to the same deviceClass. A thin-typed vg and a thick-typed vg can be used at the same time without issues as long as the thin-pool doesnt occupy 100% of the space of the vg